handler.go

```diff
@@ -144,30 +144,34 @@ func (s *handler) register(namespace string, r interface{}) {
 	for i := 0; i < val.NumMethod(); i++ {
 		method := val.Type().Method(i)
 
-		funcType := method.Func.Type()
 		hasCtx := 0
-		if funcType.NumIn() >= 2 && funcType.In(1) == contextType {
+		if method.Type.NumIn() >= 2 && method.Type.In(1) == contextType {
 			hasCtx = 1
 		}
 
 		hasRawParams := false
-		ins := funcType.NumIn() - 1 - hasCtx
+		ins := method.Type.NumIn() - 1 - hasCtx
+		opt := 0
 		recvs := make([]reflect.Type, ins)
 		for i := 0; i < ins; i++ {
 			if hasRawParams && i > 0 {
 				panic("raw params must be the last parameter")
 			}
-			if funcType.In(i+1+hasCtx) == rtRawParams {
+			if method.Type.In(i+1+hasCtx) == rtRawParams {
 				hasRawParams = true
 			}
 			recvs[i] = method.Type.In(i + 1 + hasCtx)
+			if recvs[i].Kind() == reflect.Pointer {
```

Comment:
This should only apply to trailing params - the opt couter should be reset on non pointer (non optional) args.